atc/exec/put_inputs.go

```diff
@@ -70,6 +71,45 @@ func (i specificInputs) FindAll(artifacts *artifact.Repository) ([]worker.InputS
 	return inputs, nil
 }
 
+type detectInputs struct {
+	inputs []string
+}
+
+func NewDetectInputs(params map[string]interface{}) PutInputs {
+	var inputs []string
+	for _, value := range params {
+		switch actual := value.(type) {
+		case string:
+			input := actual
+			if idx := strings.IndexByte(actual, '/'); idx >= 0 {
+				input = actual[:idx]
+			}
+			inputs = append(inputs, input)
+		default:
```

Comment:
There are two more cases to handle:

```go
map[string]interface{}
[]interface{}
```

e.g.:

```yaml
params:
  globs: [foo/*.tgz]
  some_map: {key: bar/*.tgz}
```

And this can go n levels deep, so you might need to sprinkle some recursion on this logic!